parcels/particleset.py

```diff
@@ -238,38 +254,113 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
+            if Basemap is None:
+                raise RuntimeError("Visualisation not possible: Basemap module not found!")
+            land = kwargs.get('land', False)
+            vmax = kwargs.get('vmax', None)
+            idx = self.grid.U.time_index(t)
+            U = np.array(self.grid.U.temporal_interpolate_fullfield(idx, t))
+            V = np.array(self.grid.V.temporal_interpolate_fullfield(idx, t))
+            lon = self.grid.U.lon
+            lat = self.grid.U.lat
+            if domain is not None:
+                lon = lon[lonW:lonE]
+                lat = lat[latS:latN]
+                U = U[latS:latN, lonW:lonE]
+                V = V[latS:latN, lonW:lonE]
+
+            # configuring plot
+            lat_median = np.median(lat)
+            lon_median = np.median(lon)
+            plt.figure()
+            m = Basemap(projection='merc', lat_0=lat_median, lon_0=lon_median,
+                        resolution='h', area_thresh=100,
+                        llcrnrlon=lon[0], llcrnrlat=lat[0],
+                        urcrnrlon=lon[-1], urcrnrlat=lat[-1])
+            if land:
+                m.drawcoastlines()
+                m.fillcontinents(color='burlywood')
+            parallels = np.arange(lat[0], lat[-1], abs(lat[0]-lat[-1])/5)
+            parallels = np.around(parallels, 2)
+            m.drawparallels(parallels, labels=[1, 0, 0, 0])
+            meridians = np.arange(lon[0], lon[-1], abs(lon[0]-lon[-1])/5)
+            meridians = np.around(meridians, 2)
+            m.drawmeridians(meridians, labels=[0, 0, 0, 1])
+
+            # formating velocity data for quiver plotting
+            U = np.array([U[y, x] for x in range(len(lon)) for y in range(len(lat))])
+            V = np.array([V[y, x] for x in range(len(lon)) for y in range(len(lat))])
+            speed = np.sqrt(U**2 + V**2)
+            normU = U/speed
+            normV = V/speed
+            x = np.repeat(lon, len(lat))
+            y = np.tile(lat, len(lon))
+
+            # plotting velocity vector field
+            vecs = m.quiver(x, y, normU, normV, speed, cmap=plt.cm.gist_ncar, clim=[0, vmax], scale=50, latlon=True)
+            m.colorbar(vecs, "right", size="5%", pad="2%")
+            xs, ys = m(plon, plat)
+            # plotting particle data
+            m.scatter(xs, ys, color='black')
+
+            if(self.grid.U.time_origin == 0):
+                plt.title(delta(seconds=t))
+            else:
+                plt.title(num2date(t, 'seconds since '+str(self.grid.U.time_origin)))
+
+        if savefile is None:
+            plt.show()
+            plt.pause(0.0001)
         else:
-            timestr = ' on ' + str(time_origin + delta(seconds=t))
-        plt.xlabel('Longitude')
-        plt.ylabel('Latitude')
-        plt.title('Particles' + namestr + timestr)
-        plt.show()
-        plt.pause(0.0001)
+            plt.savefig(savefile)
+            print 'Plot saved to '+savefile+'.png'
```

Comment:
This `print 'Plot saved to '+savefile+'.png'` will raise a python3 error because printing is different there. And we aim to be python3 compatible. Hence the red cross in travis..